File: CS/Converters/AlignmentConverter.cs
using System;
using System.Windows;
using System.Windows.Data;
namespace DXRibbonNotepad.Converters {
    public class AlignmentConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            return parameter.ToString() == value.ToString();
        }
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture) {
            return (bool)value ? Enum.Parse(typeof(TextAlignment), parameter.ToString()) : null;
        }
    }
}

File: VB/Converters/AlignmentConverter.vb
Imports System
Imports System.Windows
Imports System.Windows.Data

Namespace DXRibbonNotepad.Converters

    Public Class AlignmentConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Return Equals(parameter.ToString(), value.ToString())
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return If(CBool(value), [Enum].Parse(GetType(TextAlignment), parameter.ToString()), Nothing)
        End Function
    End Class
End Namespace
